src/Raven.Server/Documents/ETL/EtlProcess.cs

```diff
@@ -154,101 +158,162 @@ protected EtlProcess(Transformation transformation, TConfiguration configuration
         protected abstract IEnumerator<TExtracted> ConvertTombstonesEnumerator(IEnumerator<Tombstone> tombstones, string collection, EtlItemType type);
 
         protected abstract IEnumerator<TExtracted> ConvertCountersEnumerator(IEnumerator<CounterGroupDetail> counters, string collection);
+        
+        protected abstract IEnumerator<TExtracted> ConvertTimeSeriesEnumerator(IEnumerator<TimeSeriesSegmentEntry> timeSeries, string collection);
+        
+        protected abstract IEnumerator<TExtracted> ConvertTimeSeriesDeletedRangeEnumerator(IEnumerator<TimeSeriesDeletedRangeItem> timeSeries, string collection);
 
         protected abstract bool ShouldTrackAttachmentTombstones();
         
         public override long TaskId => Configuration.TaskId;
 
-        public virtual IEnumerable<TExtracted> Extract(DocumentsOperationContext context, long fromEtag, EtlItemType type, EtlStatsScope stats)
+        private IEnumerable<TExtracted> Extract(DocumentsOperationContext context, long fromEtag, EtlItemType type, EtlStatsScope stats)
         {
             using (var scope = new DisposableScope())
             using (var merged = new ExtractedItemsEnumerator<TExtracted>(stats, type))
             {
                 switch (type)
                 {
                     case EtlItemType.Document:
-                        var enumerators = new List<(IEnumerator<Document> Docs, IEnumerator<Tombstone> Tombstones, string Collection)>(Transformation.Collections.Count);
-
-                        if (Transformation.ApplyToAllDocuments)
-                        {
-                            var docs = Database.DocumentsStorage.GetDocumentsFrom(context, fromEtag, 0, long.MaxValue).GetEnumerator();
-                            scope.EnsureDispose(docs);
+                        ExtractDocuments(context, merged, fromEtag, stats, scope);
```

Comment:
The `merged` is immutable because it is in `using` and so cannot be used as `ref`.
I have couple of solutions to avoid that but none of them is pretty.

Reply:
Let's leave it as is then.